File: MangoAPI.BusinessLogic/Notifications/SendMessageNotification.cs
using MangoAPI.Domain.Enums;
using System;

namespace MangoAPI.BusinessLogic.Notifications;

public record SendMessageNotification(
    Guid MessageId,
    Guid UserId,
    Guid ChatId,
    string DisplayName,
    DisplayNameColour DisplayNameColour,
    string Text,
    DateTime CreatedAt,
    DateTime? UpdatedAt,
    string InReplyToUser,
    string InReplyToText,
    string AuthorImageUrl,
    string AttachmentUrl);

File: MangoAPI.Client/src/app/types/notifications/SendMessageNotification.ts
import { DisplayNameColours } from '../enums/DisplayNameColours';

export interface SendMessageNotification {
  messageId: string;
  userId: string;
  chatId: string;
  displayName: string;
  displayNameColour: DisplayNameColours;
  text: string;
  createdAt: string;
  updatedAt: string;
  inReplyToUser: string;
  inReplyToText: string;
  authorImageUrl: string;
  attachmentUrl: string;
}
